src/Authentication/Authentication.Core/Utilities/AuthenticationHelpers.cs

```diff
@@ -66,45 +66,52 @@ public static IAuthenticationProvider GetAuthProvider(IAuthContext authContext)
             switch (authContext.AuthType)
             {
                 case AuthenticationType.Delegated:
-                    {
-                        IPublicClientApplication publicClientApp = PublicClientApplicationBuilder
+                {
+                    //Specify Default RedirectUri
+                    //https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/MSAL.NET-uses-web-browser
+                    IPublicClientApplication publicClientApp = PublicClientApplicationBuilder
                         .Create(authContext.ClientId)
                         .WithTenantId(authContext.TenantId)
                         .WithAuthority(authorityUrl)
                         .WithClientCapabilities(new[] { "cp1" })
+                        .WithDefaultRedirectUri()
                         .Build();
-
-                        ConfigureTokenCache(publicClientApp.UserTokenCache, authContext);
+                    ConfigureTokenCache(publicClientApp.UserTokenCache, authContext);
+                    if (authContext.UseDeviceAuth)
```

Comment:
If you switch to enum, this will be:
```suggestion
                    if (authContext.AuthProviderType == AuthProviderType.DeviceCodeProvider)
```